src/hapi/pipelines/database/operational_presence.py

```diff
@@ -82,63 +87,67 @@ def populate(self, debug=False):
-                            admin_code=admin_code, admin_level=admin_level
-                        )
-                        org_acronym_orig = values[org_acronym_index][
-                            admin_code
-                        ][i]
-                        if not org_name_orig:
-                            org_name_orig = org_acronym_orig
+                    for i, org_str in enumerate(org_names):
+                        # * Sector processing
                         sector_orig = values[sector_index][admin_code][i]
                         # Skip rows that are missing a sector
                         if not sector_orig:
                             add_message(
                                 errors,
                                 dataset_name,
-                                f"org {org_name_orig} missing sector",
+                                f"org {org_str} missing sector",
                             )
                             continue
-                        org_type_orig = None
-                        if org_type_name_index:
-                            org_type_orig = values[org_type_name_index][
-                                admin_code
-                            ][i]
-                        country_code = admin_code
+                        sector_code = self._sector.get_sector_code(sector_orig)
+                        if not sector_code:
+                            add_missing_value_message(
+                                errors, dataset_name, "sector", sector_orig
+                            )
+                            continue
+
+                        # * Admin processing
                         if admin_level == "admintwo":
                             country_code = self._admintwo.pcode_to_iso3.get(
                                 admin_code
                             )
-                        if admin_level == "adminone":
+                        elif admin_level == "adminone":
                             country_code = self._adminone.pcode_to_iso3.get(
                                 admin_code
                             )
-                        org_info = self._org.get_org_info(
-                            org_name_orig, location=country_code
+                        else:
+                            country_code = admin_code
+                        admin2_code = admins.get_admin2_code_based_on_level(
+                            admin_code=admin_code, admin_level=admin_level
                         )
-                        org_name = org_info.get("#org+name")
-                        self._org.add_org_to_lookup(org_name_orig, org_name)
-                        org_acronym = org_info.get(
-                            "#org+acronym",
-                            values[org_acronym_index][admin_code][i],
+                        admin2_ref = self._admins.admin2_data[admin2_code]
+
+                        # * Org processing
+                        if not org_str:
```

Comment:
In this version of the code, if I'm not mistaken it appears that we use the org name to query the lookup, and then we use the acronym / org type from that if the org name is already there. We only use the acronym / org type from the raw data as a fallback. 

I thought previously we would first use the acronym / org type from the source data, since we consider orgs to be unique with name + acronym. @b-j-mills are there examples of orgs with the same name but different acronym? What about with org type (seems more likely). 

But anyway I just want to make sure I understand the logical flow.

Reply:
I think it's the same flow - first you check the canonical org table to find a match by name, and if there's a match we use the associated org type and acronym.  If not, we use the org name and acronym to match to the other orgs in the database.
I haven't seen orgs with the same name but different acronyms, but I've seen the reverse.  I have seen orgs with the same name and acronym but different org types (there's a todo in here to address this).

Reply:
Based on @b-j-mills reply above, I'll mark as resolved